src/consensus/aft/raft.h

```diff
@@ -538,11 +546,30 @@ namespace aft
       return offset;
     }
 
+    bool same_node_ids(
+      const kv::Configuration::Nodes& conf1,
+      const kv::Configuration::Nodes& conf2)
+    {
+      if (conf1.size() == conf2.size())
+      {
+        for (const auto& [nid, _] : conf1)
+        {
+          if (conf2.find(nid) == conf2.end())
+          {
+            return false;
+          }
+        }
+        return true;
+      }
+      return false;
+    }
```

Comment:
Since, the `hostname` and `port` for a given node should always stay the same, isn't this just `conf1 == conf2`?

Reply:
Yes, except there's no `operator==` for `NodeInfo`. I just wanted to compare the node ids, so I opted not to add an operator. Do you think it would be better to add one?

Reply:
`raft.h` is already chunky so defining the `operator==` for `NodeInfo` is probably simpler.